viper.py

```diff
@@ -724,6 +750,38 @@ def fit_chunk(o, chunk, obsname, targ=None, tpltarg=None):
     print(file=parunit)
     #vpr.plot_rvo(rv, e_rv)
 
+
+if createtpl:
+    wtpl_new = {}
+    tpl_new = {}
+    for o in orders:
+        gplot.reset()
+        gplot.key("title 'order: %s' noenhance" % (o))
+        wt = np.array(list(tpl_all[o,0].values())) 	# wavelength
+        ft = np.array(list(tpl_all[o,1].values()))	# data
+        mt = np.array(list(tpl_all[o,2].values()))	# weighting
+        mt[np.isnan(ft)] = np.nan
+        mt[ft<0] = np.nan
```

Comment:
Weight (`mt`) should be not be `nan`. Nan or bad flux should have zero weight `mt=0`.

Reply:
On a first test it worked out fine. I will include it and have an eye on it during future tests.